### PR TITLE
Allow Rails 7 in gemspec

### PR DESCRIPTION
**Pull request by chrisccerami**

### CODE REVIEW

**Review by amancevice — APPROVED** [review]

File: lib/replica_pools/query_cache.rb [hunk 1]
```diff
@@ -19,12 +19,10 @@ module QueryCache
...
     # through to whatever connection is current.
     def select_all(*args)
-      # there may be more args for Rails 5.0+, but we only care about arel, name, and binds for caching.
       relation, name, raw_binds = args
+      arel = arel_from_relation(relation)
 
-      # Rails 6.2 breaks this method as locked? is no longer available
-      # https://github.com/kickstarter/replica_pools/issues/26
-      if !query_cache_enabled || locked?(relation)
+      if !query_cache_enabled && !(arel.respond_to?(:locked) && arel.locked)
```
Comment (chrisccerami) [inline lib/replica_pools/query_cache.rb:25]: I think this isn't quite right, based on the test failure. I based it off of [the Rails 7 version of this method](https://github.com/rails/rails/blob/v7.0.3.1/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L96)

Comment (chrisccerami) [reply lib/replica_pools/query_cache.rb]: I decided to pull back in the `locked?` method from Rails 6.1.

File: lib/replica_pools/query_cache.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require 'pry'
```
Comment (amancevice) [inline lib/replica_pools/query_cache.rb:1]: remnant from debugging?

**Review by amancevice — APPROVED** [review]
possible errant require 'pry' but otherwise 👍

**Comment by amancevice** [comment]
@chrisccerami I'll release this now

**Pull request merged**